jcl/src/java.base/share/classes/java/lang/Class.java

```diff
@@ -3282,10 +3283,19 @@ private AnnotationCache getAnnotationCache() {
 		if (annotationsData == null) {
 			annotationCacheResult = new AnnotationCache(null, buildAnnotations(null));
 		} else {
+			byte[] classPtrBytes = new byte[VM.FJ9OBJECT_SIZE];
+			for (int i = 0; i < VM.FJ9OBJECT_SIZE; ++i) {
```

Comment:
there is a more direct way of doing this. You can use `long classPtr = getUnsafe.getLong(annotationsData, Unsafe.ARRAY_BYTE_BASE_OFFSET + ((annotationsData.length * Unsafe.ARRAY_BYTE_INDEX_SCALE) -  VM.FJ9OBJECT_SIZE);`

You should double check my math, but thats the idea

Reply:
Why is the native layout reversed from the java layout? I.e. I'm curious why the least significant byte of the pointer was first (and why I needed to put the last bytes first to get the correct class pointer)

Reply:
Its the mechanics of Little endian. If you write a mutlibyte word (ie. `*classPtrAddr = declaringClass;`), the least significant byte will be placed first. If you attempt to read that word byte-by-byte, you need to take that into accout. However, if you read that word as a multibyte byte word (ie `getLong1`), it just works out because the read takes endianness into account.

Reply:
In my example, Ive used `getLong`, you need to use the correct size based on the size of the header.